pkg/rehearse/jobs.go

```diff
@@ -29,34 +32,26 @@ const (
 	rehearseLabel = "ci.openshift.org/rehearse"
 )
 
-type prowJobClientWithDry struct {
-	pj.ProwJobInterface
-
-	dry bool
-}
-
 // NewProwJobClient creates a ProwJob client with a dry run capability
 func NewProwJobClient(clusterConfig *rest.Config, namespace string, dry bool) (pj.ProwJobInterface, error) {
+	if dry {
+		pjcset := pjclientsetfake.NewSimpleClientset()
+		pjcset.Fake.PrependReactor("create", "prowjobs", func(action testing.Action) (bool, runtime.Object, error) {
+			pj := action.(testing.CreateAction).GetObject().(*pjapi.ProwJob)
+			jobAsYAML, err := yaml.Marshal(pj)
+			if err != nil {
+				return true, nil, fmt.Errorf("failed to marshal job to YAML: %v", err)
+			}
+			fmt.Printf("%s\n", jobAsYAML[:10])
+			return false, nil, nil
+		})
+		return pjcset.ProwV1().ProwJobs(namespace), nil
+	}
```

Comment:
From an earlier (failed) attempt at solving the problem, this has the benefit of not even requiring a kubeconfig in dry-run mode.

Unfortunately, it would require an unhealthy amount of hacks to fake the jobs transitioning from pending to a finished state.